bindings/http/http_test.go

```diff
@@ -147,7 +147,163 @@ func TestDefaultBehavior(t *testing.T) {
+}
+
+func TestNon2XXErrorsSuppressed(t *testing.T) {
+	handler := NewHTTPHandler()
+	s := httptest.NewServer(handler)
+	defer s.Close()
+
+	hs, err := InitBinding(s, map[string]string{"errorIfNot2XX": "false"})
+	require.NoError(t, err)
+	verifyNon2XXErrorsSuppressed(t, hs, handler)
+}
+
+func InitBindingForHTTPS(s *httptest.Server, extraProps map[string]string) (bindings.OutputBinding, error) {
+	m := bindings.Metadata{Base: metadata.Base{
+		Properties: map[string]string{
+			"url": s.URL,
+		},
+	}}
+	for k, v := range extraProps {
+		m.Properties[k] = v
+	}
+	hs := bindingHttp.NewHTTP(logger.NewLogger("test"))
+	err := hs.Init(m)
+	return hs, err
+}
+
+func httpsHandler(w http.ResponseWriter, r *http.Request) {
+	// r.TLS gets ignored by HTTP handlers.
+	// in case where client auth is not required, r.TLS.PeerCertificates will be empty.
+	res := fmt.Sprintf("%v", len(r.TLS.PeerCertificates))
+	io.WriteString(w, res)
+}
+
+func TestDefaultBehaviorHTTPS(t *testing.T) {
+	handler := NewHTTPHandler()
+	server := setupHTTPSServer(t, true, handler)
+	defer server.Close()
+
+	certMap := map[string]string{
+		"MTLSEnable":     "true",
+		"MTLSRootCA":     filepath.Join(".", "testdata", "ca.pem"),
+		"MTLSClientCert": filepath.Join(".", "testdata", "client.pem"),
+		"MTLSClientKey":  filepath.Join(".", "testdata", "client.key"),
+	}
+	hs, err := InitBindingForHTTPS(server, certMap)
+	require.NoError(t, err)
+
+	verifyDefaultBehaviors(t, hs, handler)
+}
+
+func TestNon2XXErrorsSuppressedHTTPS(t *testing.T) {
+	handler := NewHTTPHandler()
+	server := setupHTTPSServer(t, true, handler)
+	defer server.Close()
+
+	certMap := map[string]string{
+		"MTLSEnable":     "true",
+		"MTLSRootCA":     filepath.Join(".", "testdata", "ca.pem"),
+		"MTLSClientCert": filepath.Join(".", "testdata", "client.pem"),
+		"MTLSClientKey":  filepath.Join(".", "testdata", "client.key"),
+		"errorIfNot2XX":  "false",
+	}
+	hs, err := InitBindingForHTTPS(server, certMap)
+	require.NoError(t, err)
+	verifyNon2XXErrorsSuppressed(t, hs, handler)
+}
+
+func TestHTTPSBinding(t *testing.T) {
+	handler := http.NewServeMux()
+	handler.HandleFunc("/testhttps", httpsHandler)
+	server := setupHTTPSServer(t, true, handler)
+	defer server.Close()
+	t.Run("get with https with valid client cert and clientAuthEnabled true", func(t *testing.T) {
+		certMap := map[string]string{
+			"MTLSEnable":     "true",
+			"MTLSRootCA":     filepath.Join(".", "testdata", "ca.pem"),
+			"MTLSClientCert": filepath.Join(".", "testdata", "client.pem"),
+			"MTLSClientKey":  filepath.Join(".", "testdata", "client.key"),
+		}
+		hs, err := InitBindingForHTTPS(server, certMap)
+		require.NoError(t, err)
+
+		req := TestCase{
+			input:      "GET",
+			operation:  "get",
+			metadata:   map[string]string{"path": "/testhttps"},
+			path:       "/testhttps",
+			err:        "",
+			statusCode: 200,
+		}.ToInvokeRequest()
+		response, err := hs.Invoke(context.TODO(), &req)
+		assert.Nil(t, err)
```

Comment:
Nit:

```suggestion
		assert.NoError(t, err)
```

Better semantics also in case oferrors

Reply:
(Same below many times)